pkg/cassandra/datacenter.go

```diff
@@ -103,6 +104,7 @@ type DatacenterConfig struct {
 	Users                 []cassdcapi.CassandraUser
 	PodTemplateSpec       *corev1.PodTemplateSpec
 	MgmtAPIHeap           *resource.Quantity
+	SoftPodAntiAffinity   *bool
```

Comment:
Do we need a pointer here? I think it would be simpler to just use the value type.

Reply:
Talked with @Miles-Garnsey offline and the pointer is indeed needed here.

Reply:
We need to use a pointer to distinguish false from unset here so that users can override the cluster-level setting with the DC level setting. 

If the cluster level is true but DC level is false, we need to ensure that the DC-level false value is really false, not just unset.